learn/work/sparse-matrix/index.html.md

```diff
@@ -0,0 +1,316 @@
+
+
+
+
+::: {.cell layout-align="center"}
+
+```{.r .cell-code}
+reviews_tbl <- reviews_tbl |>
+  mutate(SCORE = factor(SCORE, levels = c(1, 0), labels = c("great", "other")))
+```
+:::
+
+
+
+
+Since `reviews_tbl` is now a tibble, we can use `initial_split()` as we usually do.
+
+
+
+
+::: {.cell layout-align="center"}
+
+```{.r .cell-code}
+set.seed(1234)
+
+review_split <- initial_split(reviews_tbl)
+review_train <- training(review_split)
+review_test <- testing(review_split)
+
+review_folds <- vfold_cv(review_train)
+```
+:::
+
+
+
+
+Next, we will specify our workflow. Since we are showcasing how sparse data works in tidymodels, we will stick to a simple lasso regression model. These models tend to work well with sparse predictors. `penalty` has been set to be tuned.
+
+
+
+
+::: {.cell layout-align="center"}
+
+```{.r .cell-code}
+rec_spec <- recipe(SCORE ~ ., data = review_train)
+
+lm_spec <- logistic_reg(penalty = tune()) |>
+  set_engine("glmnet")
+
+wf_spec <- workflow(rec_spec, lm_spec)
+```
+:::
+
+
+
+
+With everything in order, we can now evaluate several different values of `penalty` with `tune_grid()`.
+
+
+
+
+::: {.cell layout-align="center"}
+
+```{.r .cell-code}
+tune_res <- tune_grid(wf_spec, review_folds)
+```
+:::
+
+
+
+
+Despite the size of the data, this code runs quite quickly due to the sparse encoding of the data. Once the tuning process is done, then we can look at the performance for different values of regularization.
+
+
+
+
+::: {.cell layout-align="center"}
+
+```{.r .cell-code}
+autoplot(tune_res)
+```
+
+::: {.cell-output-display}
+![](figs/autoplot-1.svg){fig-align='center' width=672}
+:::
+:::
+
+
+
+
+We can now finalize the workflow and fit the final model on the training data set.
+
+
+
+
+::: {.cell layout-align="center"}
+
+```{.r .cell-code}
+wf_final <- finalize_workflow(
+ wf_spec, 
+  select_best(tune_res, metric = "roc_auc")
+ )
+
+wf_fit <- fit(wf_final, review_train)
+```
+:::
+
+
+
+
+With this fitted model, we can now predict with a sparse tibble.
+
+
+
+
+::: {.cell layout-align="center"}
+
+```{.r .cell-code}
+predict(wf_fit, review_test)
+#> # A tibble: 3,750 × 1
+#>    .pred_class
+#>    <fct>      
+#>  1 other      
+#>  2 great      
+#>  3 great      
+#>  4 great      
+#>  5 great      
+#>  6 great      
+#>  7 great      
+#>  8 great      
+#>  9 great      
+#> 10 great      
+#> # ℹ 3,740 more rows
+```
+:::
+
+
+
+
+## Session information {#session-info}
+
+
+
+
+::: {.cell layout-align="center"}
+
+```
+#> ─ Session info ─────────────────────────────────────────────────────
+#>  setting  value
+#>  version  R version 4.4.0 (2024-04-24)
+#>  os       macOS 15.0
+#>  system   aarch64, darwin20
+#>  ui       X11
+#>  language (EN)
+#>  collate  en_US.UTF-8
+#>  ctype    en_US.UTF-8
+#>  tz       America/Los_Angeles
+#>  date     2024-10-14
+#>  pandoc   2.17.1.1 @ /opt/homebrew/bin/ (via rmarkdown)
+#> 
+#> ─ Packages ─────────────────────────────────────────────────────────
```

Comment:
Lots of dev versions and remotes here. Fine for now, but a TODO for once these are sent out to CRAN. :)